postinstall.js

```diff
@@ -267,6 +286,31 @@ async function setupStatueSSG(options = {}) {
         }
       }
 
+      if (!isDefaultTemplate) {
+        const templatePackageJsonPath = path.join(templateDir, 'package.json');
+        if (fs.existsSync(templatePackageJsonPath)) {
+          const templatePkg = JSON.parse(fs.readFileSync(templatePackageJsonPath, 'utf8'));
+          if (templatePkg.dependencies) {
+            packageJson.dependencies = packageJson.dependencies || {};
+            for (const [dep, version] of Object.entries(templatePkg.dependencies)) {
+              if (!packageJson.dependencies[dep]) {
+                packageJson.dependencies[dep] = version;
+                templateDependenciesAdded = true;
```

Comment:
I don't think we should have special logic for resolving template - default versioning mismatches, I could understand adding missing dependencies or removing them when changing from template->default or template->template locally though